File: aoc-2022/day-06/python/part1_sol.py
f = open("../input.txt")
data = f.read().rstrip()


# data = "zcfzfwzzqfrljwzlrfnpqdbhtmscgvjw"

for i in range(3, len(data)):
    s = set()
    for c in data[i-3:i+1]:
        s.add(c)
    if len(s) == 4:
        print(i+1)
        break

File: aoc-2022/day-06/rust/part1_sol.rs
use std::fs;
use std::collections::HashSet;

fn main() {
    let data = fs::read_to_string("../input.txt").expect("Couldn't read from file");
    for i in 4..data.len() {
        let mut set = HashSet::new();
        for c in data[i-4..i].chars() {
            set.insert(c);
        }
        if set.len() == 4 {
            println!("{}", i);
            break;
        }
    }
}